localstack/services/dynamodb/dynamodb_listener.py

```diff
@@ -319,11 +327,21 @@ def return_response(self, method, path, data, headers, response):
                 stream_spec = dynamodb_get_table_stream_specification(table_name=table_name)
                 if stream_spec:
                     record['dynamodb']['StreamViewType'] = stream_spec['StreamViewType']
+
         elif action == 'BatchWriteItem':
-            records = self.prepare_batch_write_item_records(record, data)
+            records, unprocessed_items = self.prepare_batch_write_item_records(record, data)
             for record in records:
                 event_sources_or_streams_enabled = (event_sources_or_streams_enabled or
                     has_event_sources_or_streams_enabled(record['eventSourceARN'], streams_enabled_cache))
+            if response.status_code == 200 and len(unprocessed_items) > 0:
+                content = json.loads(to_str(response.content))
+                table_name = list(data['RequestItems'].keys())[0]
+                if table_name not in content['UnprocessedItems']:
+                    content['UnprocessedItems'][table_name] = []
```

Comment:
The proper response is:

```json
"UnprocessedItems": {
  "<TableName>": {
     "DeleteRequest":{
        "Key1": {<Key1>},
        "Key2": {<Key2>}
     },
     "PutRequest": {
        "Item1": { Item1 },
        "Item2": { Item2 }
     }
  }
}
```

Reply:
Sorry but acording to https://docs.aws.amazon.com/amazondynamodb/latest/APIReference/API_BatchWriteItem.html#DDB-BatchWriteItem-response-UnprocessedItems the right syntax would be

```
   "UnprocessedItems": { 
      "string" : [ 
         { 
            "DeleteRequest": { 
               "Key": { 
```

So a list, not a JSON object